packages/transaction-controller/src/types.ts

```diff
@@ -1556,6 +1561,9 @@ export type BatchTransactionParams = {
   /** Address of the target contract or EOA. */
   to?: Hex;
 
+  /** Type of the transaction. */
+  type?: TransactionType;
```

Comment:
Rather than adding this to this parameter specific type, should we instead change `batchTransactions` on `TransactionMeta` to be an array of `NestedTransactionMetadata`?

As that is already this type and the transaction type and it keeps the parameters themselves separate.

We could rename `NestedTransactionMetadata` in future but not now, to avoid the breaking change.